### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other changes, this commit ensures the generated HTML doc for
HexDocs.pm will become the main source doc for this Elixir library which
leverage on ExDoc features.

### CODE REVIEW

File: .formatter.exs
Comment (whatyouhide) [inline .formatter.exs:1]: Let's remove this, `.formatter.exs` is standard in all Elixir projects so no need to specify this here.

```suggestion
```

File: .gitignore
Comment (whatyouhide) [inline .gitignore:2]: Unnecessary change. Please revert all these trailing slashes.

File: .gitignore
Comment (whatyouhide) [inline .gitignore:10]: Unnecessary change, please revert :)

File: .gitignore
Comment (whatyouhide) [inline .gitignore:5]: Unnecessary change

File: .gitignore
Comment (whatyouhide) [inline .gitignore:26]: I ran tests and nothing generated anything in `/tmp`. Please remove this.

File: .gitignore
Comment (whatyouhide) [inline .gitignore:29]: This is not a QuickCheck artifact, is it? It's the protoc compiler plugin

File: README.md
Comment (whatyouhide) [inline README.md:8]: Please revert all of these changes, they are unnecessary in the README. Thanks!

File: README.md
Comment (whatyouhide) [inline README.md:196]: Did @tony612 request this change? Let's drop it otherwise.

File: lib/protobuf/protoc/context.ex
Comment (whatyouhide) [inline lib/protobuf/protoc/context.ex:7]: Please revert the changes in this file, the new formatting is not the one supported by the formatter so there is no reason in reformatting this file :)

File: mix.exs
Comment (whatyouhide) [inline mix.exs:4]: The repo has been renamed :)

```suggestion
  @source_url "https://github.com/elixir-protobuf/protobuf"
```

File: mix.exs
Comment (whatyouhide) [inline mix.exs:67]: This is an unnecessary change. Can you revert it?

File: mix.exs
Comment (whatyouhide) [inline mix.exs:76]: Nice!

**Review by whatyouhide — CHANGES_REQUESTED** [review]

File: README.md
Comment (kianmeng) [reply README.md:196]: > Did @tony612 request this change? Let's drop it otherwise.

Nope and I'll remove it for now. But good to put copyright and license details in the readme. 

Our of curiosity, looking at the GitHub repo URL change, is this Elixir library now under the co-maintenance between @tony612 and Elixir core team?

File: .gitignore
Comment (kianmeng) [reply .gitignore:26]: > I ran tests and nothing generated anything in `/tmp`. Please remove this.

Noted. FYI, the `.gitignore` was copied and merged into existing `.gitignore` from generated file from `mix new foobar` using latest Elixir version.

File: .gitignore
Comment (kianmeng) [reply .gitignore:29]: > This is not a QuickCheck artifact, is it? It's the protoc compiler plugin

I will put under `# Misc.`.

File: README.md
Comment (kianmeng) [reply README.md:8]: Noted. I will just put the GitHub Action badge.

File: lib/protobuf/protoc/context.ex
Comment (kianmeng) [reply lib/protobuf/protoc/context.ex]: > Please revert the changes in this file, the new formatting is not the one supported by the formatter so there is no reason in reformatting this file :)

Formatting as in Markdown formatting or Protobuf formatting?

**Comment by whatyouhide** [comment]
Awesome job, thanks @kianmeng! Can you rebase this on master? Once you do, we'll be good to merge!

**Comment by kianmeng** [comment]
> Awesome job, thanks @kianmeng! Can you rebase this on master? Once you do, we'll be good to merge!

@whatyouhide Done. Please check

**Comment by whatyouhide** [comment]
Thanks @kianmeng! 💟

**Pull request merged**
